proposals/proposal_1/lib/main.dart

```diff
@@ -0,0 +1,44 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:proposal_1/src/app.dart';
+import 'package:proposal_1/src/items/item_controller.dart';
+import 'package:proposal_1/src/items/item_service.dart';
+import 'package:proposal_1/src/logging/debug_logging_service.dart';
+import 'package:proposal_1/src/logging/profile_logging_service.dart';
+import 'package:proposal_1/src/logging/release_logging_service.dart';
+import 'package:proposal_1/src/settings/settings_controller.dart';
+import 'package:proposal_1/src/settings/settings_service.dart';
+
+void main() {
+  // Set up cross-cutting Services, such as Logging, Analytics, or Feature Flags
+
+  // First, set up the correct LoggingService depending on the environment
+  final loggingService = kProfileMode
+      ? ProfileLoggingService()
+      : kReleaseMode
+          ? ReleaseLoggingService()
+          : DebugLoggingService();
+
+  // Catch Flutter errors and forward them to the logging service.
+  FlutterError.onError = (FlutterErrorDetails details) {
+    loggingService.error(details.exceptionAsString(), details.stack);
+  };
+
+  // Catch Dart errors and forward them to the logging service.
+  runZonedGuarded(() async {
+    // Setup the Settings Service
+    final settingsController = SettingsController(SettingsService());
+
+    // Load the user's settings before displaying the App.
+    await settingsController.loadSettings();
```

Comment:
As with everything, it depends :)

If you're storing the settings on disk, I'd say you almost always want to load those setting before running the app. Otherwise you may have a "Flash of Unthemed Content" -- meaning you may show the app in the default theme, then load the user settings very quickly from disk while showing a spinner, then change everything to a different theme. That can be a pretty quick and jarring transition, especially if you're going from a default light theme to a dark theme, or vice-versa.

Instead, it's generally better from a user experience standpoint to let the Splash screen show for an additional 5-10ms to load the settings from disk, then show the app with the correct theme from the get-go. This avoids any big theme changes.

However, if you're loading settings from the network, this technique is not ideal.

Reply:
You're right. Let's keep with this. You can always update settings from the network later (and have FOUC then) but generally, this makes sense.